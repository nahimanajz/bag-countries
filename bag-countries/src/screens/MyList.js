import { Bin } from "../assets/Bin";
import { Check } from "../assets/Check";
import { SearchIcon } from "../assets/SearchIcon";

export function MyList({children}){
    return(
            <div className="flex-col">
                <div className="flex-wrap">
                    <div className="grid-col-2">
                        <span> <SearchIcon /> </span>                      
                      <input text="name" type="text"  placeholder="Search for A Country"/>
                    </div>
                    <div>
                        <select>
                            <option>Filter by region</option>
                            <option value="value">Africa</option>
                            <option>America</option>
                            <option>Europe</option>
                        </select>
                        
                    </div>
                </div>
                <div className="grid-col-3">
                    {/* Map countries from api */}
                    <div className="grid-row-4">
                        <div>
                            <img className="flag"
                                 src="https://upload.wikimedia.org/wikipedia/commons/1/17/Flag_of_Rwanda.svg" 
                                alt="flag"></img>
                            </div>
                        <div className="subtitle2 p-h">Country name</div>
                        <div className="p-h">
                            <ul>
                                <li>Population:{   12.9  }</li>
                                <li>Capital: {'kGALI'} </li>
                                <li>Currency: {'Rwf'}</li>
                            </ul>
                        </div>
                        <div className="flex-end p-h">
                            <span><Bin /></span>
                            <span><Check /></span>
                        </div>
                    </div>
                    <div className="grid-row-4">
                        <div>
                            <img className="flag"
                                 src="https://upload.wikimedia.org/wikipedia/commons/1/17/Flag_of_Rwanda.svg" 
                                alt="flag"></img>
                            </div>
                        <div className="subtitle2 p-h">Country name</div>
                        <div className="p-h">
                            <ul>
                                <li>Population:{   12.9  }</li>
                                <li>Capital: {'kGALI'} </li>
                                <li>Currency: {'Rwf'}</li>
                            </ul>
                        </div>
                        <div className="flex-end p-h">
                            <span><Bin /></span>
                            <span><Check /></span>
                        </div>
                    </div>
                    <div className="grid-row-4">
                        <div>
                            <img className="flag"
                                 src="https://upload.wikimedia.org/wikipedia/commons/1/17/Flag_of_Rwanda.svg" 
                                alt="flag"></img>
                            </div>
                        <div className="subtitle2 p-h">Country name</div>
                        <div className="p-h">
                            <ul>
                                <li>Population:{   12.9  }</li>
                                <li>Capital: {'kGALI'} </li>
                                <li>Currency: {'Rwf'}</li>
                            </ul>
                        </div>
                        <div className="flex-end p-h">
                            <span><Bin /></span>
                            <span><Check /></span>
                        </div>
                    </div>
                    
                </div>
                <div> list of countrie</div>
            </div>

    )
}