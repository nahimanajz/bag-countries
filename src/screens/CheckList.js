import axios from "axios";
import { useState } from "react";
import { toast, ToastContainer } from "react-toastify";
import { CountryFilter } from "../components/CountryFilter";
import { ListableCountry } from "../components/ListableCountry";
import { toLowerCase, BACKEND_API_ROUTE } from "../util";

/**
 * Loop countries on user UI screens
 * @param {Array} countries fetched countries from App component to pass to Listable item
 * @param {Object} userInfo signed user
 @return {Component, Component} Listable Country filter
 */

export function CheckList({ countries, userInfo, ...props }) {
  const [myList, setMyList] = useState();
  const [regionCountries, setRegionCountries] = useState();

  const addCountryToList = (e) => {
    const country = countries.filter(({ name }) =>
      toLowerCase(name).includes(toLowerCase(e.target.value))
    );
    setMyList(country); 
    setRegionCountries();
  };
  const addCountryByRegion = async (e) => {
    const country = countries.filter(
      ({ region }) => toLowerCase(region) === toLowerCase(e.target.value)
    );
    setRegionCountries(country);
    setMyList(country);
  };
  //Todo: addTo list in backend
  const handleCheckList = async (name) => {
   
    const { data } = await axios.post(`${BACKEND_API_ROUTE}/add/country`, {
      name,
      user: userInfo._id,
    });
    if (data) {
      toast(data.message);
    } else {
      toast("Something is wrong");
    }
  };

  return (
    <>
      <ToastContainer />
      <CountryFilter
        addCountryToList={addCountryToList}
        addCountryByRegion={addCountryByRegion}
      />
      <ListableCountry
        goToDetail={props.goToDetail}
        add={handleCheckList}
        countries={myList ? myList : regionCountries && regionCountries}
      />
    </>
  );
}
