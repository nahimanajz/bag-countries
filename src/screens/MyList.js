import { useEffect, useState } from "react";
import { CountryFilter } from "../components/CountryFilter";
import { ListableCountry } from "../components/ListableCountry";
import { toLowerCase } from "../util";
/**
 * - Countries to Visit  
 * @param {Function} changeScreenTitle to change screen to appropriat visited screen
 * @param {Function} updateCountry  to updated selected country to visited list
 * @param {Function} deleteCountry  to remove country from visit list
 * @param {Array} countries fetched countries from App component to pass to Listable item
 * @param {*} JSX tags to append on my list screen
 */

export function MyList({
  countries,
  children,
  changeScreenTitle,
  updateCountry,
  deleteCountry,
  ...props
}) {
  const [myList, setMyList] = useState();
  const [regionCountries, setRegionCountries] = useState();
  const [countryItem, setCountryItem] = useState();
  
  const addCountryToList = (e) => {
    const country = countries.filter(({ name }) =>
      toLowerCase(name).includes(toLowerCase(e.target.value))
    );
    setMyList(country);
    setRegionCountries();
  };
  const addCountryByRegion = async (e) => {
    const country = countries.filter(
      ({ region }) => toLowerCase(region) === toLowerCase(e.target.value)
    );
    setRegionCountries(country);
    setMyList(country);
  };
  useEffect(() => {
    setCountryItem(myList ? myList : regionCountries && regionCountries);
  }, [myList, regionCountries]);

  return (
    <div className="flex-col">
      <CountryFilter
        addCountryToList={addCountryToList}
        addCountryByRegion={addCountryByRegion}
      />
      <ListableCountry
        countries={countryItem}
        goToDetail={props.goToDetail}
        updateCountry={updateCountry}
        deleteCountry={deleteCountry}
      />
    </div>
  );
}
