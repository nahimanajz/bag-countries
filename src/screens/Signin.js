import axios from 'axios';
import { useCallback, useState } from 'react'
import {useNavigate } from "react-router-dom";
import { Mail } from "../assets/Mail";
import { BACKEND_API_ROUTE } from "../util";
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';

export default function Signin(props) {

   const[state, setState] = useState({email:'', password:''})
   const handleChange = useCallback(
    (e) => {
      setState({...state, [e.target.name]: e.target.value});
    },
    [state],
  );
   


   const navigate = useNavigate();
    const handleSignin =  async()=>{     
        const{email, password} = state;  
        const{ data}  = await axios.post(`${BACKEND_API_ROUTE}/signin`,{email, password})
        if(!data.message){   
            navigate('/my-list',{loggedUser:data})
            props.showDashboard(true)
            props.userInfo(data)
           
        }
        toast(data.message);
        }
       
    return(
        <>
            <ToastContainer />
            <form method="post">          
		      <ul className="modal">
                 <li>
                    <label>Email</label>        
                       <input type="email" onChange={handleChange} placeholder="Email" name="email" required />
                  </li>  
                  <li>
                        <label>Password</label>        
                        <input type="password" onChange={handleChange} placeholder="Password" name="password" required/>
                  </li>
                  <li>                     
                     <input  type="button"
                        onClick={handleSignin} 
                        className="btn" 
                      value="Signin" style={{height: 48}} 
                     /> 
                  </li>      
              </ul>
            </form>
        </>
            
    )
}