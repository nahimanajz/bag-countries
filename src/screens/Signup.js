import { useCallback, useState } from 'react'
import axios from 'axios'
import { BACKEND_API_ROUTE } from '../util'
import { toast, ToastContainer } from 'react-toastify';

/**
 * - User signin 
 * @param {Function} props.showDashboard to show main dashboard later on sinup
 * @param {Function} props.userInfo  setUser info so that can view relavant info on dashboard
 */

export function Signup({ showDashboard, userInfo}){
   const[state, setState] = useState({name:'', email:'', password:'',phone:'',dob:'',country:''})
   const handleChange = useCallback(
    (e) => {
      setState({...state, [e.target.name]: e.target.value});
    },
    [state],
  );
   const handleSignup= async ()=>{
   const {name, email, password,phone,dob,country} = state;
   const{ data}  = await axios.post(`${BACKEND_API_ROUTE}/signup`,{name, email, password,phone,dob,country})
  

    if(data.error){ 
       toast('Please correct your data');

    }else{
      toast('Well done!!')
      userInfo(data.newUser);
      showDashboard(true)
      
    }
    
   }

    return(
         <>
            <ToastContainer />
            <form method="post">
		      <ul className="modal">
               <li>
                  <label>Full Names</label>        
                  <input type="text" onChange={handleChange} placeholder="Full Names" name="name" required />
                      
                  </li>
                  <li>
                        <label>Email</label>        
                        <input type="email" onChange={handleChange} placeholder="Email" name="email" required />
                  </li>  
                  <li>
                        <label>Password</label>        
                        <input type="password" onChange={handleChange} placeholder="Password" name="password" required/>
                  </li> 
                  <li>
                        <label>Phone</label>        
                        <input type="tel" onChange={handleChange} placeholder="Phone" name="phone" required/>
                  </li> 
                  <li>
                     <label>DOB</label>
                     <input type="date" onChange={handleChange} placeholder="Date Of Birth" name="dob" required/>
                  </li> 
                 
                  <li>
                     <label>Country</label>
                     <input type="text" onChange={handleChange} placeholder="Country" name="country" required/>
                     
                  </li>
                  <li>                     
                     <input  type="button"
                        onClick={handleSignup} 
                        className="btn" 
                      value="Signup" style={{height: 48}} 
                     disabled={state.length <6?true: false}/> 
                  </li>    
               
               </ul>
                </form>
         </>
    )
}