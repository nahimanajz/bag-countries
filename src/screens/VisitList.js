
import { ListableCountry } from "../components/ListableCountry";

export default function VisitList({countries, goToDetail, updateCountry, deleteCountry}){

   return (
       <>
      <ListableCountry
        countries={countries}
        goToDetail={goToDetail}
        updateCountry={updateCountry}
        deleteCountry={deleteCountry}
        
      />
      </>
   )
}