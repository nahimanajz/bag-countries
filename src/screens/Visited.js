import { ListableCountry } from "../components/ListableCountry";

export default function Visited({countries, goToDetail, updateCountry, deleteCountry}){

    return(
            <ListableCountry
            countries={countries}
            goToDetail={goToDetail}
            updateCountry={updateCountry}
            deleteCountry={deleteCountry}
            checked={true}
        />
        )
}