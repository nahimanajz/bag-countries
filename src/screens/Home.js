import { useState } from "react";
import Signin  from './Signin';
import { Signup } from "./Signup";

/**
 * Toggle Login and Signup screens
 * @param {Function} props.setShowDashoard to show main dashboard later on logging in or successfull signup
 */

export default function Home(props) {
    const[showLogin, setShowLogin] = useState(false);
    const switchScreens = ()=> setShowLogin(!showLogin);

    return(<>
    
        <div className="cover">
            <div className="model-header">
                <div className={showLogin?'inactive':'active'} onClick={switchScreens}>Signup</div>
                <div className={showLogin?'active':'inactive'} onClick={switchScreens}>Signin</div>
            </div>
            {showLogin? <Signin showDashboard={props.setShowDashboard} userInfo={props.userInfo}/>:<Signup showDashboard={props.setShowDashboard} userInfo={props.userInfo}/> }
             
        </div>
        
        </>
    )
}
