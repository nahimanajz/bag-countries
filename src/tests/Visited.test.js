// ComponentName.test.js
import React from 'react';
import { shallow } from 'enzyme';
import VisitList from '../screens/VisitList';
import Visited from '../screens/Visited';
import { MyList } from '../screens/MyList';
import { CheckList } from '../screens/CheckList';

describe("render dashboard screens", () => {
  it("should render visit list  component without error", () => {
    const wrapper = shallow(<VisitList />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
  it("should render visited component without error", () => {
    const wrapper = shallow(<Visited />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
  it("should render My list  component without error", () => {
    const wrapper = shallow(<MyList />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
  it("should render check list   without error", () => {
    const wrapper = shallow(<CheckList />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});