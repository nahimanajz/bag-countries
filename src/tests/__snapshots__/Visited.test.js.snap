// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` should render visited component without error 1`] = `
Array [
  <React.Fragment>
    <ListableCountry />
  </React.Fragment>,
]
`;

exports[`ComponentName should render visited component without error 1`] = `
Array [
  <React.Fragment>
    <ListableCountry />
  </React.Fragment>,
]
`;

exports[`render dashboard screens should render My list  component without error 1`] = `
Array [
  <div
    className="flex-col"
  >
    <CountryFilter
      addCountryByRegion={[Function]}
      addCountryToList={[Function]}
    />
    <ListableCountry />
  </div>,
]
`;

exports[`render dashboard screens should render check list   without error 1`] = `
Array [
  <React.Fragment>
    <ToastContainer
      autoClose={5000}
      closeButton={[Function]}
      closeOnClick={true}
      draggable={true}
      draggableDirection="x"
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      role="alert"
      rtl={false}
      theme="light"
      transition={[Function]}
    />
    <CountryFilter
      addCountryByRegion={[Function]}
      addCountryToList={[Function]}
    />
    <ListableCountry
      add={[Function]}
    />
  </React.Fragment>,
]
`;

exports[`render dashboard screens should render visit list  component without error 1`] = `
Array [
  <React.Fragment>
    <ListableCountry />
  </React.Fragment>,
]
`;

exports[`render dashboard screens should render visited component without error 1`] = `
Array [
  <ListableCountry
    checked={true}
  />,
]
`;

exports[`render screens should render My list  component without error 1`] = `
Array [
  <React.Fragment>
    <ListableCountry />
  </React.Fragment>,
]
`;

exports[`render screens should render check list   component without error 1`] = `
Array [
  <React.Fragment>
    <ToastContainer
      autoClose={5000}
      closeButton={[Function]}
      closeOnClick={true}
      draggable={true}
      draggableDirection="x"
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      role="alert"
      rtl={false}
      theme="light"
      transition={[Function]}
    />
    <CountryFilter
      addCountryByRegion={[Function]}
      addCountryToList={[Function]}
    />
    <ListableCountry
      add={[Function]}
    />
  </React.Fragment>,
]
`;

exports[`render screens should render check list   without error 1`] = `
Array [
  <React.Fragment>
    <ToastContainer
      autoClose={5000}
      closeButton={[Function]}
      closeOnClick={true}
      draggable={true}
      draggableDirection="x"
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      role="alert"
      rtl={false}
      theme="light"
      transition={[Function]}
    />
    <CountryFilter
      addCountryByRegion={[Function]}
      addCountryToList={[Function]}
    />
    <ListableCountry
      add={[Function]}
    />
  </React.Fragment>,
]
`;

exports[`render screens should render viited component without error 1`] = `
Array [
  <React.Fragment>
    <ListableCountry />
  </React.Fragment>,
]
`;

exports[`render screens should render visit component without error 1`] = `
Array [
  <React.Fragment>
    <ListableCountry />
  </React.Fragment>,
]
`;

exports[`render screens should render visit list  component without error 1`] = `
Array [
  <React.Fragment>
    <ListableCountry />
  </React.Fragment>,
]
`;

exports[`render screens should render visited component without error 1`] = `
Array [
  <React.Fragment>
    <ListableCountry />
  </React.Fragment>,
]
`;

exports[`render screens should render visited component without error 2`] = `
Array [
  <React.Fragment>
    <ListableCountry />
  </React.Fragment>,
]
`;

exports[`render screens should render visited component without error 3`] = `
Array [
  <React.Fragment>
    <ListableCountry />
  </React.Fragment>,
]
`;
